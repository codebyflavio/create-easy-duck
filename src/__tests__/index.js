import {
  createTypes,
  createActionCreators,
  createReducer,
  createSelectors,
  createDuck,
} from '../../dist'

test('should create base types', () => {
  expect(createTypes('RANDOM_NAME')).toMatchSnapshot()
  expect(createTypes('ANOTHER_NAME')).toMatchSnapshot()
})

test('should create action creators', () => {
  const action = createActionCreators('RANDOM_NAME')
  expect(action.start()).toMatchSnapshot()
  expect(action.finish({ hello: 'there' })).toMatchSnapshot()
  expect(action.error(new Error('An error occurred'))).toMatchSnapshot()
  expect(action.error()).toMatchSnapshot()
  expect(action.reset()).toMatchSnapshot()
})

test('should create reducer', () => {
  const TYPE = 'RANDOM_NAME'
  expect(createReducer(TYPE)()).toMatchSnapshot()

  const action = createActionCreators(TYPE)
  const reducer = createReducer(TYPE)
  let state = reducer()
  expect(state).toMatchSnapshot()

  state = reducer(state, action.start())
  expect(state).toMatchSnapshot()

  state = reducer(state, action.finish({ it: 'works' }))
  expect(state).toMatchSnapshot()

  state = reducer(state, action.error())
  expect(state).toMatchSnapshot()

  state = reducer(state, action.reset())
  expect(state).toMatchSnapshot()
})

test('should create selectors', () => {
  const selectors = createSelectors('RANDOM_NAME', {
    custom: state => (state.isLoading ? 'Loading' : 'Loaded'),
  })
  const state = {
    isLoading: false,
    payload: { data: { hello: 'there' } },
    error: 'It failed!',
  }

  expect(selectors.getAll(state)).toMatchSnapshot()
  expect(selectors.getIsLoading(state)).toMatchSnapshot()
  expect(selectors.getPayload(state)).toMatchSnapshot()
  expect(selectors.getPayloadData(state)).toMatchSnapshot()
  expect(selectors.getError(state)).toMatchSnapshot()
  expect(selectors.custom(state)).toMatchSnapshot()
  expect(selectors.custom({ RANDOM_NAME: state })).toMatchSnapshot()
})

test('should create duck', () => {
  const duck = createDuck({
    type: 'RANDOM_NAME',
    initialState: {
      isLoading: false,
      error: null,
      payload: { data: { count: 1 } },
    },
    selectors: {
      getCount: state => state.payload.data.count,
    },
  })

  let state = duck.reducer()

  expect(duck.selectors.getAll(state)).toMatchSnapshot()
  expect(duck.selectors.getIsLoading(state)).toMatchSnapshot()
  expect(duck.selectors.getPayload(state)).toMatchSnapshot()
  expect(duck.selectors.getPayloadData(state)).toMatchSnapshot()
  expect(duck.selectors.getError(state)).toMatchSnapshot()
  expect(duck.selectors.getCount(state)).toMatchSnapshot()
})
