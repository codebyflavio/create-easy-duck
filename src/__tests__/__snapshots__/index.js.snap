// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create action creators 1`] = `
Object {
  "type": "RANDOM_NAME/START",
}
`;

exports[`should create action creators 2`] = `
Object {
  "payload": Object {
    "hello": "there",
  },
  "type": "RANDOM_NAME/FINISH",
}
`;

exports[`should create action creators 3`] = `
Object {
  "error": "An error occurred",
  "type": "RANDOM_NAME/ERROR",
}
`;

exports[`should create action creators 4`] = `
Object {
  "error": "Something went wrong",
  "type": "RANDOM_NAME/ERROR",
}
`;

exports[`should create action creators 5`] = `
Object {
  "type": "RANDOM_NAME/RESET",
}
`;

exports[`should create base types 1`] = `
Object {
  "ERROR": "RANDOM_NAME/ERROR",
  "FINISH": "RANDOM_NAME/FINISH",
  "RESET": "RANDOM_NAME/RESET",
  "START": "RANDOM_NAME/START",
}
`;

exports[`should create base types 2`] = `
Object {
  "ERROR": "ANOTHER_NAME/ERROR",
  "FINISH": "ANOTHER_NAME/FINISH",
  "RESET": "ANOTHER_NAME/RESET",
  "START": "ANOTHER_NAME/START",
}
`;

exports[`should create duck 1`] = `
Object {
  "error": null,
  "isLoading": false,
  "payload": Object {
    "data": Object {
      "count": 1,
    },
  },
}
`;

exports[`should create duck 2`] = `false`;

exports[`should create duck 3`] = `
Object {
  "data": Object {
    "count": 1,
  },
}
`;

exports[`should create duck 4`] = `
Object {
  "count": 1,
}
`;

exports[`should create duck 5`] = `null`;

exports[`should create duck 6`] = `1`;

exports[`should create reducer 1`] = `
Object {
  "error": null,
  "isLoading": false,
  "payload": null,
}
`;

exports[`should create reducer 2`] = `
Object {
  "error": null,
  "isLoading": false,
  "payload": null,
}
`;

exports[`should create reducer 3`] = `
Object {
  "error": null,
  "isLoading": true,
  "payload": null,
}
`;

exports[`should create reducer 4`] = `
Object {
  "error": null,
  "isLoading": false,
  "payload": Object {
    "it": "works",
  },
}
`;

exports[`should create reducer 5`] = `
Object {
  "error": "Something went wrong",
  "isLoading": false,
  "payload": null,
}
`;

exports[`should create reducer 6`] = `
Object {
  "error": null,
  "isLoading": false,
  "payload": null,
}
`;

exports[`should create selectors 1`] = `
Object {
  "error": "It failed!",
  "isLoading": false,
  "payload": Object {
    "data": Object {
      "hello": "there",
    },
  },
}
`;

exports[`should create selectors 2`] = `false`;

exports[`should create selectors 3`] = `
Object {
  "data": Object {
    "hello": "there",
  },
}
`;

exports[`should create selectors 4`] = `
Object {
  "hello": "there",
}
`;

exports[`should create selectors 5`] = `"It failed!"`;

exports[`should create selectors 6`] = `"Loaded"`;

exports[`should create selectors 7`] = `"Loaded"`;
